### ISSUE TITLE
Cannot find environment property file 'docker-host-env-dev.properties'

### ISSUE THREAD

**Issue by rishikeshpalve** [TIBCOSoftware/bw6-plugin-maven#58]
Trying to run Maven Build from Studio to build docker image and start the docker container.

docker:start goal fails due to error - 
Failed to execute goal io.fabric8:docker-maven-plugin:0.14.2:start (default-cli) on project com.tibco.bwce.banq.account.service.application: Execution default-cli of goal io.fabric8:docker-maven-plugin:0.14.2:start failed: Failed to create contained configuration for [tibco/bwce/banq]: Cannot find environment property file 'docker-host-env-dev.properties'

Workaround:
- Copy 'docker-host-env-dev.properties' file to %%bw.application%%.parent 
OR
- Run docker:build goal from %%bw.application%%.parent directory and then run docker:start goal from %%bw.application%%

**Environment details**
Maven Plugin version: 1.2.0

Maven version: 2.3.1

Product : TIBCO Businessworks Container edition

Product version: 2.2.0

**Debug Output**

ERROR] DOCKER> Error occurred during container startup, shutting down...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] com.tibco.bwce.banq.account.service.application.parent SUCCESS [  0.641 s]
[INFO] com.tibco.bwce.banq.account.service ................ SUCCESS [  1.151 s]
[INFO] com.tibco.bwce.banq.account.service.application .... FAILURE [  7.346 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 10.326 s
[INFO] Finished at: 2017-02-09T06:03:16-05:00
[INFO] Final Memory: 26M/385M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.14.2:start (default-cli) on project com.tibco.bwce.banq.account.service.application: Execution default-cli of goal io.fabric8:docker-maven-plugin:0.14.2:start failed: Failed to create contained configuration for [tibco/bwce/banq]: Cannot find environment property file 'docker-host-env-dev.properties' -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.fabric8:docker-maven-plugin:0.14.2:start (default-cli) on project com.tibco.bwce.banq.account.service.application: Execution default-cli of goal io.fabric8:docker-maven-plugin:0.14.2:start failed: Failed to create contained configuration for [tibco/bwce/banq]
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal io.fabric8:docker-maven-plugin:0.14.2:start failed: Failed to create contained configuration for [tibco/bwce/banq]
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 20 more
Caused by: java.lang.IllegalArgumentException: Failed to create contained configuration for [tibco/bwce/banq]
	at io.fabric8.maven.docker.service.RunService.createContainerConfig(RunService.java:246)
	at io.fabric8.maven.docker.service.RunService.createAndStartContainer(RunService.java:96)
	at io.fabric8.maven.docker.StartMojo.executeInternal(StartMojo.java:85)
	at io.fabric8.maven.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:164)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	... 21 more
Caused by: java.lang.IllegalArgumentException: Cannot find environment property file 'docker-host-env-dev.properties'
	at io.fabric8.maven.docker.access.ContainerCreateConfig.addPropertiesFromFile(ContainerCreateConfig.java:169)
	at io.fabric8.maven.docker.access.ContainerCreateConfig.environment(ContainerCreateConfig.java:66)
	at io.fabric8.maven.docker.service.RunService.createContainerConfig(RunService.java:236)
	... 25 more
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :com.tibco.bwce.banq.account.service.application
[DEBUG] Connection manager is shutting down
[DEBUG] http-outgoing-0: Close connection
[DEBUG] http-outgoing-0: Close connection
[DEBUG] Connection manager shut down

**Comment by RajibBiswasTibco** [comment]
@rishikeshpalve , All commands should be run from %%bw.application%%.parent directory.

**Comment by rishikeshpalve** [comment]
@RajibBiswasTibco I had faced this issue while trying to run commands from %%bw.application%%.parent directory. I had to copy 'docker-host-env-dev.properties' file into this directory to make it work.

**Comment by SonaliDasgupta** [comment]
@rishikeshpalve Please try this out with the latest version of BW6 Maven plugin, in which the Fabric8 plugin and Docker Maven plugin been upgraded. If the error persists, could you please share the project?

**Comment by rishikeshpalve** [comment]
Tried out with latest version on Maven plug-in. The issue is resolved now.

**Comment by vtaront** [comment]
Hello,

I have also this issue on plugin **2.0.1**.

In the pom.xml of the application module, the properties file is referenced:
Property:
`<docker.env.property.file>docker-host-env-dev.properties</docker.env.property.file>`

On configuration of properties-maven-plugin plugin:
```
<groupId>org.codehaus.mojo</groupId>
<artifactId>properties-maven-plugin</artifactId>
<version>1.0.0</version>
[...]
<configuration>
	<files>
		<file>${docker.property.file}</file>
		<file>${docker.env.property.file}</file>
	</files>
</configuration>
```

And finaly on run parameter of fabric8 plugin:
```
<groupId>io.fabric8</groupId>
<artifactId>docker-maven-plugin</artifactId>
<version>0.26.1</version>
[...]
<envPropertyFile>${docker.env.property.file}</envPropertyFile>
```

Firstly, I build successfully the container with:
`clean package initialize docker:build`

Then, I try to start it, but unsuccessfully:
`initialize docker:start`


[INFO] ------------------------------------------------------------------------
[INFO] Building ems-bwce 1.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- properties-maven-plugin:1.0.0:read-project-properties (default) @ ems-bwce ---
[INFO] 
[INFO] --- docker-maven-plugin:0.26.1:start (default-cli) @ ems-bwce ---
[ERROR] DOCKER> Error occurred during container startup, shutting down...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] ems-bwce.parent .................................... SUCCESS [  0.406 s]
[INFO] ems-bwce.module .................................... SUCCESS [  0.397 s]
[INFO] ems-bwce ........................................... FAILURE [  0.478 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.804 s
[INFO] Finished at: 2018-10-12T08:17:51Z
[INFO] Final Memory: 26M/305M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.26.1:start (default-cli) on project ems-bwce: Execution default-cli of goal io.fabric8:docker-maven-plugin:0.26.1:start failed: Failed to create contained configuration for [bwce/ems-bwce]: Cannot find environment property file 'docker-host-env-dev.properties' -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :ems-bwce
Build step 'Invoke top-level Maven targets' marked build as failure
Finished: FAILURE

**Comment by SonaliDasgupta** [comment]
@vtaront Could you please share your project ? Since at my end, only the docker-dev.properties file is referenced in the POM tags, on generating POM for application.

**Comment by vtaront** [comment]
Actually this file is generated when you add environment variables: "Env vars" field in "Docker host run configuration", second screen when you "Generate POM for Application".

[ems-bwce.zip](https://github.com/TIBCOSoftware/bw6-plugin-maven/files/2472338/ems-bwce.zip)

**Comment by SonaliDasgupta** [comment]
@awagle @abhidetibco Please refer to Pull Request #195 for resolution of this issue, and also issue #134 . The error was due to the absolute location of the docker environment properties file not being found by the docker maven plugin. It can be solved by including the full path of the file in the generated application POM.

**Comment by SonaliDasgupta** [comment]
Resolved from Release 2.0.2 onwards.
